tests/test_pre_commit_conf_generator.py

```diff
@@ -0,0 +1,298 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+import pathlib
+import tempfile
+import types
+import unittest
+
+from lsst.ts.pre_commit_conf.pre_commit_conf_generator import (
+    FLAKE8_CONFIG_FILE,
+    ISORT_CONFIG_FILE,
+    MYPY_CONFIG_FILE,
+    PRE_COMMIT_CONFIG_FILE_NAME,
+    copy_config_files,
+    generate_pre_commit_conf_file,
+    parse_args,
+    update_dot_gitignore,
+)
+
+ROOT = pathlib.Path(__file__)
+
+# Directories with data used by this script.
+TEMPLATES_DIR = ROOT.resolve().parents[0] / "templates"
+
+# Data dir.
+DATA_DIR = ROOT.resolve().parents[0] / "data"
+
+# pre-commit config files for validation.
+PRE_COMMIT_CONF_COMPLETE = DATA_DIR / "pre_commit_conf_complete.yaml"
+PRE_COMMIT_CONF_WITHOUT_ISORT = DATA_DIR / "pre_commit_conf_without_isort.yaml"
+PRE_COMMIT_CONF_WITHOUT_ISORT_AND_MYPY = (
+    DATA_DIR / "pre_commit_conf_without_isort_and_mypy.yaml"
+)
+PRE_COMMIT_CONF_WITHOUT_MYPY = DATA_DIR / "pre_commit_conf_without_mypy.yaml"
+
+# .gitignore files for validation.
+DOT_GITIGNORE_COMPLETE = DATA_DIR / "dot_gitignore_complete"
+DOT_GITIGNORE_WITHOUT_ISORT = DATA_DIR / "dot_gitignore_without_isort"
+DOT_GITIGNORE_WITHOUT_ISORT_AND_MYPY = DATA_DIR / "dot_gitignore_without_isort_and_mypy"
+DOT_GITIGNORE_WITHOUT_MYPY = DATA_DIR / "dot_gitignore_without_mypy"
+
+logging.basicConfig(
+    format="%(asctime)s:%(levelname)s:%(name)s:%(message)s", level=logging.DEBUG
+)
+
+
+class PrecommitConfGeneratorTestCase(unittest.IsolatedAsyncioTestCase):
+    async def test_parse_args_no_command_line_args(self) -> None:
+        args = parse_args(command_line_args=[])
+        assert args.no_isort is False
+        assert args.no_mypy is False
+        assert args.dest == "."
+
+        args = parse_args(command_line_args=["--no-isort"])
+        assert args.no_isort is True
+        assert args.no_mypy is False
+        assert args.dest == "."
+
+        args = parse_args(command_line_args=["--no-mypy"])
+        assert args.no_isort is False
+        assert args.no_mypy is True
+        assert args.dest == "."
+
+        args = parse_args(command_line_args=["--dest", "/tmp"])
+        assert args.no_isort is False
+        assert args.no_mypy is False
+        assert args.dest == "/tmp"
+
+        args = parse_args(command_line_args=["--no-isort", "--no-mypy"])
+        assert args.no_isort is True
+        assert args.no_mypy is True
+        assert args.dest == "."
+
+        args = parse_args(
+            command_line_args=["--no-isort", "--no-mypy", "--dest", "/tmp"]
+        )
+        assert args.no_isort is True
+        assert args.no_mypy is True
+        assert args.dest == "/tmp"
+
+    def validate_pre_commit_conf(
+        self, args: types.SimpleNamespace, expected_conf_file: pathlib.Path
+    ) -> None:
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            # Set destination path to avoid left over files from running the
```

Comment:
I think you mean "ruining" instead of "running"? Please consider "breaking" or "interfering with" or...

Reply:
No, I mean `running`. The unit tests are run which create files and I am using a temp dir to avoid adding files to the project.

Reply:
That makes sense.

In what way does setting the destination path alter the fact that this is a temp dir that will go away when the test is finished running?

Reply:
It doesn't. The temp dir will go away so that saves cleanup code.